File: .cpp/Program/Mouse.cpp
#pragma once
#include <bits/stdc++.h>
#include "Device.cpp"
using namespace std;

class Mouse : public Device {
    private:
        string connectionType;
        int DPI;
        string sensorType;

    public:
        Mouse () { }

        Mouse (string brand, string model, string connectionType, int DPI, string sensorType) : Device (brand, model) {
            this->connectionType = connectionType;
            this->DPI = DPI;
            this->sensorType = sensorType;
        }

        void setConnectionType (string connectionType) { this->connectionType = connectionType; }
        void setDPI (int DPI) { this->DPI = DPI; }
        void setSensorType (string sensorType) { this->sensorType = sensorType; }

        string getConnectionType () { return connectionType; }
        int getDPI () { return DPI; }
        string getSensorType () { return sensorType; }

        void info () override {
            cout << "Mouse" << endl;
            cout << "| Brand             : " << getBrand() << endl;
            cout << "| Model             : " << getModel() << endl;
            cout << "| Connection Type   : " << connectionType << endl;
            cout << "| DPI               : " << DPI << endl;
            cout << "| Sensor Type       : " << sensorType << endl;
        }

        ~Mouse () { }
};

File: .py/Program/Mouse.py
from Device import Device

class Mouse (Device):
    __connectionType = ""
    __DPI = 0
    __sensorType = ""

    def __init__ (self, brand = "", model = "", connectionType = "", DPI = 0, sensorType = ""):
        super ().__init__ (brand, model)
        self.__connectionType = connectionType
        self.__DPI = DPI
        self.__sensorType = sensorType

    def setConnectionType (self, connectionType):
        self.__connectionType = connectionType

    def setDPI (self, DPI):
        self.__DPI = DPI

    def setSensorType (self, sensorType):
        self.__sensorType = sensorType

    def getConnectionType (self):
        return self.__connectionType
    
    def getDPI (self):
        return self.__DPI
    
    def getSensorType (self):
        return self.__sensorType
    
    def info (self):
        print("| Mouse")
        print(f"| Brand             : {self.getBrand()}")
        print(f"| Model             : {self.getModel()}")
        print(f"| Connection Type   : {self.getConnectionType()}")
        print(f"| DPI               : {self.getDPI()}")
        print(f"| Sensor Type       : {self.getSensorType()}")